### PR TITLE
#120 & #145

### PR DESCRIPTION
**Pull request by dserkowski**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -167,42 +168,52 @@ <h3 class="sr-only">Math editor</h3>
...
+                                    <br>
+                                    <h3 class="sr-only">Clean up and add new step</h3>
+                                    <button id='addStep' class="btn btn-primary pointer editorButton" data-toggle="tooltip"
+                                            title="Clean up the cross outs and start a new step (⌨: shift+enter)"
+                                            onclick="AddStep();"
+                                            data-step="3"
+                                            data-intro="Clean-up your work and start a new step."><i
+                                            class="fa fa-plus"></i> Add Step
+                                    </button>
+                                    <div hidden id="updateControls">
```
Comment (pawelcieszko) [inline index.html]: add sr-only classes to increase accesibility

File: index.html [hunk 2]
```diff
@@ -167,42 +168,52 @@ <h3 class="sr-only">Math editor</h3>
...
-                                        title="Cancel edit" onclick="ExitUpdate();"
-                                        data-step="3"
-                                        data-intro="Cancel edit."><i
-                                        class="fa fa-ban" ></i> Cancel edit
-                                    </button>                        
+                            <div id="mathPalette" class="pl-0 d-flex flex-nowrap" ></div>
+                            <div id="mathButtons" class="d-flex flex-nowrap justify-content-between">
+                                <div>
+                                    <br>
+                                    <button id="scratch-pad-button" class="btn btn-default btn-scratch btn-background paletteButton" data-toggle="tooltip" title="Display/hide sketch pad">
```
Comment (pawelcieszko) [inline index.html]: add sr-only classes to increase accesibility

File: scripts/scripts.js [hunk 3]
```diff
@@ -524,14 +669,14 @@ function NewRowOrRowsAfterCleanup(mathContent) {
 }
 
 function UpdateRowAfterCleanup(mathContent, mathStepNumber) {
-    
+
```
Comment (pawelcieszko) [inline scripts/scripts.js]: remove undesired blankline

File: index.html [hunk 4]
```diff
@@ -167,42 +168,52 @@ <h3 class="sr-only">Math editor</h3>
...
-                                        class="fa fa-pencil"></i> Update Step
-                                    </button>                         
-                                    <br/>
-                                    <button id='cancelEdit' class="btn btn-default pointer editorButton" data-toggle="tooltip"
-                                        title="Cancel edit" onclick="ExitUpdate();"
-                                        data-step="3"
-                                        data-intro="Cancel edit."><i
-                                        class="fa fa-ban" ></i> Cancel edit
-                                    </button>                        
+                            <div id="mathPalette" class="pl-0 d-flex flex-nowrap" ></div>
```
Comment (pawelcieszko) [inline index.html]: remove undesired whitespace

File: scripts/scripts.js [hunk 5]
```diff
@@ -41,11 +41,129 @@ $(document).ready(function(){
...
+        availableLineWidths: [1,2,4,8,16],
+        availableEraserWidths: [1,2,4,8,16,64],
+        availableFontSizes: [1,2,4,8,16],
+        colorScheme: {
+            main: 'rgba(0,98,217,0.6)',
+            control: 'white',
+            activeControl: 'rgba(230,230,230,1)',
+            inputText: 'black'
+        }
+
```
Comment (pawelcieszko) [inline scripts/scripts.js]: I think this line is not needed

File: index.html (hunk 1 above)
Comment (pawelcieszko) [reply index.html]: done

File: index.html [hunk 6]
```diff
@@ -167,42 +168,54 @@ <h3 class="sr-only">Math editor</h3>
...
-                                    </button>                         
-                                    <br/>
-                                    <button id='cancelEdit' class="btn btn-default pointer editorButton" data-toggle="tooltip"
-                                        title="Cancel edit" onclick="ExitUpdate();"
-                                        data-step="3"
-                                        data-intro="Cancel edit."><i
-                                        class="fa fa-ban" ></i> Cancel edit
-                                    </button>                        
+                            <div id="mathPalette" class="pl-0 d-flex flex-nowrap"></div>
+                            <div id="mathButtons" class="d-flex flex-nowrap justify-content-between">
```
Comment (oskarhinc) [inline index.html]: change the id to be more relevant.

File: index.html [hunk 7]
```diff
@@ -167,42 +168,54 @@ <h3 class="sr-only">Math editor</h3>
...
+                                            data-step="3"
+                                            data-intro="Clean-up your work and start a new step."><i
+                                            class="fa fa-plus"></i> Add Step
+                                    </button>
+                                    <div hidden id="updateControls">
+                                        <h3 class="sr-only">Update the step</h3>
+                                        <button id='updateStep' class="btn btn-success pointer editorButton" data-toggle="tooltip"
+                                            title="Update the step (⌨: shift+enter)"
+                                            data-step="3"
+                                            data-intro="Update the step."><i
```
Comment (oskarhinc) [inline index.html]: Please move this mark to the new line.

File: index.html [hunk 8]
```diff
@@ -167,42 +168,54 @@ <h3 class="sr-only">Math editor</h3>
...
+                                <div>
+                                    <button class="btn btn-undo btn-background paletteButton" id="undoDelete" data-toggle="tooltip"
+                                            title="Undo Last Delete" onclick="UndoDeleteStep()" style="margin-bottom: 5px;"></button>
+                                    <br>
+                                    <h3 class="sr-only">Clean up and add new step</h3>
+                                    <button id='addStep' class="btn btn-primary pointer editorButton" data-toggle="tooltip"
+                                            title="Clean up the cross outs and start a new step (⌨: shift+enter)"
+                                            onclick="AddStep();"
+                                            data-step="3"
+                                            data-intro="Clean-up your work and start a new step."><i
```
Comment (oskarhinc) [inline index.html]: Please move this mark to the new line.

File: index.html [hunk 9]
```diff
@@ -167,42 +168,54 @@ <h3 class="sr-only">Math editor</h3>
...
+                                            data-step="3"
+                                            data-intro="Update the step."><i
+                                            class="fa fa-pencil"></i> Update Step
+                                        </button>
+                                        <br/>
+                                        <h3 class="sr-only">Cancel edit</h3>
+                                        <button id='cancelEdit' class="btn btn-default pointer editorButton" data-toggle="tooltip"
+                                            title="Cancel edit" onclick="ExitUpdate();"
+                                            data-step="3"
+                                            data-intro="Cancel edit."><i
```
Comment (oskarhinc) [inline index.html]: Please move this mark to the new line.

File: scripts/scripts.js [hunk 10]
```diff
@@ -41,11 +41,129 @@ $(document).ready(function(){
                 var file = $('#fileid').get(0).files[0];
                 readBlob();
         });
+    InitScratchPad();
     ShowWorkArea(false);
 
     $('#undoDelete').hide();
 });
 
+function InitScratchPad() {
```
Comment (oskarhinc) [inline scripts/scripts.js]: Please consider storing the configuration of this to another file.

File: scripts/scripts.js [hunk 11]
```diff
@@ -41,11 +41,129 @@ $(document).ready(function(){
...
+            activeControl: 'rgba(230,230,230,1)',
+            inputText: 'black'
+        }
+    });
+
+    ScratchPadPainterro.show();
+    $('#scratch-pad-containter').hide();
+    $('#scratch-pad-button').click(function() {
+        $('#scratch-pad-containter').slideToggle("fast", function() {
+            if ($("#scratch-pad-containter").is(":visible") === true) {
```
Comment (oskarhinc) [inline scripts/scripts.js]: is this needed (=== true)?

File: scripts/scripts.js [hunk 12]
```diff
@@ -41,11 +41,129 @@ $(document).ready(function(){
...
+            }
+        });
+    });
+
+    padPallete = $('#scratch-pad-containter-bar > div > span').first();
+    padPallete.append('<button id="clear-button" type="button" class="ptro-icon-btn ptro-color-control" title="Clear the scratch pad"><i class="ptro-icon ptro-icon-close"></i></button>');
+    $('#clear-button').click(function() {
+        ClearAndResizeScrachPad();
+    });
+    $('.ptro-icon-btn').css('border-radius','.25rem');
```
Comment (oskarhinc) [inline scripts/scripts.js]: whitespace between parameters

File: scripts/scripts.js [hunk 13]
```diff
@@ -41,11 +41,129 @@ $(document).ready(function(){
...
+}
+
+function ClearAndResizeScrachPad() {
+    ScratchPadPainterro.clear();
+}
+
+function ClearScrachPad() {
+    ScratchPadPainterro.clearBackground();
+    ScratchPadPainterro.worklog.current = null;
+    ScratchPadPainterro.worklog.clean = true; // it is because Painterro displays a modal if we want to replace an existing ScratchPad content
```
Comment (oskarhinc) [inline scripts/scripts.js]: Please put a comment in a line above.

File: styles/styles.css [hunk 14]
```diff
@@ -487,9 +487,36 @@ pulse.off {
...
+    width: 330px;
+}
+
+.btn-scratch {
+    background-color: white;
+    background-image: url(../images/scratch.png);
+}
+
+.btn-scratch:active {
+    background-image: url(../images/scratch.png)!important;
```
Comment (oskarhinc) [inline styles/styles.css]: Wouldn't it work without overriding, despite it's :active?

File: index.html [hunk 15]
```diff
@@ -32,6 +32,7 @@
     <script src="scripts/scripts.js"></script>
```
Comment (oskarhinc) [inline index.html]: Please update debig/indexs.html as well

File: scripts/scripts.js (hunk 10 above)
Comment (pawelcieszko) [reply scripts/scripts.js]: ok

File: index.html (hunk 15 above)
Comment (pawelcieszko) [reply index.html]: debug*

File: scripts/scripts.js (hunk 13 above)
Comment (pawelcieszko) [reply scripts/scripts.js]: ok

File: scripts/scripts.js (hunk 12 above)
Comment (pawelcieszko) [reply scripts/scripts.js]: ok

File: index.html (hunk 7 above)
Comment (pawelcieszko) [reply index.html]: Rigth, I have also noticed that there is used an improper markup <i>. I will change it to <span>

File: index.html (hunk 8 above)
Comment (pawelcieszko) [reply index.html]: Right, I have also noticed that there is used an improper markup <i>. I will change it to <span>

File: index.html (hunk 9 above)
Comment (pawelcieszko) [reply index.html]: Right, I have also noticed that there is used an improper markup <i>. I will change it to <span>

File: styles/styles.css (hunk 14 above)
Comment (pawelcieszko) [reply styles/styles.css]: ok I will delete it

File: scripts/scripts.js (hunk 11 above)
Comment (pawelcieszko) [reply scripts/scripts.js]: not really, corrected

File: index.html (hunk 15 above)
Comment (pawelcieszko) [reply index.html]: I think we should update it after all changes are commited.

Comment (dserkowski) [reply index.html]: Done

File: scripts/painterroConfiguration.js [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+    availableLineWidths: [1,2,4,8,16],
+    availableEraserWidths: [1,2,4,8,16,64],
+    availableFontSizes: [1,2,4,8,16],
+    colorScheme: {
+        main: 'rgba(0,98,217,0.6)',
+        control: 'white',
+        activeControl: 'rgba(230,230,230,1)',
+        inputText: 'black'
+    }
+};
```
Comment (oskarhinc) [inline scripts/painterroConfiguration.js]: no newline

File: scripts/painterroConfiguration.js [hunk 17]
```diff
@@ -0,0 +1,85 @@
+var painterroConfiguration = {
```
Comment (oskarhinc) [inline scripts/painterroConfiguration.js]: It looks like a script, but it fits more to the data package, I think.

File: scripts/painterroConfiguration.js (hunk 16 above)
Comment (pawelcieszko) [reply scripts/painterroConfiguration.js]: there is a new line

File: scripts/painterroConfiguration.js (hunk 17 above)
Comment (pawelcieszko) [reply scripts/painterroConfiguration.js]: it may be. I will move it to data directory

**Pull request merged**